src/tests/redux/cartoSlice.test.js

```diff
@@ -0,0 +1,183 @@
+import * as cartoSlice from '../../redux/cartoSlice';
+import { mockAppStoreConfiguration } from './mockReducerManager';
+import { VOYAGER } from 'src/basemaps';
+
+const INITIAL_STATE = {
+  viewState: {
+    latitude: 0,
+    longitude: 0,
+    zoom: 1,
+    pitch: 0,
+    bearing: 0,
+    dragRotate: false
+  }
+};
+
+describe('carto slice', () => {
+  const store = mockAppStoreConfiguration();
+  store.reducerManager.add('carto', cartoSlice.createCartoSlice(INITIAL_STATE));
+
+  describe('source actions', () => {
+    const sourceInfo = {
+      id: 'source-test-id',
+      data: 'SELECT * FROM stores',
+      type: 'sql'
+    };
+
+    test('should add a new source', () => {
+      store.dispatch(cartoSlice.addSource(sourceInfo));
+      const { carto } = store.getState();
+      expect(carto.dataSources[sourceInfo.id]).toEqual(sourceInfo);
+    });
+
+    test('should remove a source', () => {
+      store.dispatch(cartoSlice.removeSource(sourceInfo.id));
+      const { carto } = store.getState();
+      expect(carto.dataSources).not.toHaveProperty(sourceInfo.id);
+    });
+  });
+
+  describe('layer actions', () => {
+    const layerInfo = {
+      id: 'whatever-id',
+      source: 'whatever-source-id'
+    };
+
+    const extraInfo = {
+      id: layerInfo.id,
+      layerAttributes: { a: 1 }
+    };
+
+    test('should not update a layer info if there is no layer', () => {
+      store.dispatch(cartoSlice.updateLayer(extraInfo));
+      const { carto } = store.getState();
+      expect(carto.layers).not.toHaveProperty(layerInfo.id);
+    });
+
+    test('should add a new layer', () => {
+      store.dispatch(cartoSlice.addLayer(layerInfo));
+      const { carto } = store.getState();
+      expect(carto.layers[layerInfo.id]).toEqual(layerInfo);
+    });
+
+    test('should update a layer with extra layerAttributes info', () => {
+      store.dispatch(cartoSlice.updateLayer(extraInfo));
+      const { carto } = store.getState();
+      expect(carto.layers[layerInfo.id]).toEqual({
+        ...layerInfo,
+        ...extraInfo.layerAttributes
+      });
+    });
+
+    test('should remove a layer', () => {
+      store.dispatch(cartoSlice.removeLayer(layerInfo.id));
+      const { carto } = store.getState();
+      expect(carto.layers).not.toHaveProperty(layerInfo.id);
+    });
+  });
+
+  describe('basemap actions', () => {
+    test('should update with a new basemap', () => {
+      store.dispatch(cartoSlice.setBasemap(VOYAGER));
+      const { carto } = store.getState();
+      expect(carto.basemap).toBe(VOYAGER);
```

Comment:
👍🏻  This test would have saved us some debugging hours (we had an issue with baseMap vs basemap some days ago)